packages/docusaurus/src/commands/deploy.ts

```diff
@@ -97,18 +97,23 @@ export default async function deploy(
     (projectName.indexOf('.github.io') !== -1 ? 'master' : 'gh-pages');
   console.log(`${chalk.cyan('deploymentBranch:')} ${deploymentBranch}`);
 
+  const useSSH = process.env.USE_SSH;
+
   const githubHost =
     process.env.GITHUB_HOST || siteConfig.githubHost || 'github.com';
+  const githubPort = process.env.GITHUB_PORT || siteConfig.githubPort;
 
-  const useSSH = process.env.USE_SSH;
   const gitPass: string | undefined = process.env.GIT_PASS;
   let gitCredentials = `${gitUser}`;
   if (gitPass) {
     gitCredentials = `${gitCredentials}:${gitPass}`;
   }
 
-  const sshRemoteBranch: string = `git@${githubHost}:${organizationName}/${projectName}.git`;
-  const nonSshRemoteBranch: string = `https://${gitCredentials}@${githubHost}/${organizationName}/${projectName}.git`;
+  const githubAddress = `${githubHost}${githubPort ? ':' + githubPort : ''}`;
+  const githubPath = `${organizationName}/${projectName}`;
+
+  const sshRemoteBranch: string = `ssh://git@${githubAddress}/${githubPath}.git`;
```

Comment:
Can you make it so that not adding a port is 100% retrocompatible with the URL we had before?

It looks to me it is not retrocompatible:

Before: `git@${githubHost}:${organizationName}/${projectName}.git`;
After: `ssh://git@${githubAddress}/${githubPath}.git` which expands to `ssh://git@${githubHost}/${organizationName}/${projectName}.git`

I'd like this change to have benefits for you, but be 100% safe for all other users. Not providing a port should lead to the  same final URL.

I'd appreciate if you could extract this URL construction logic in a fn and add tests to this specific part, as it becomes more complicated to maintain

Reply:
Note, the url on Github UI has no port but still contains `:`, so  it looks like we probably want to keep it too?

![image](https://user-images.githubusercontent.com/749374/113722603-dda7a100-96f0-11eb-9a0e-e926f23ae707.png)

Reply:
Hi @slorber,

Nice. I isolated this logic and I created tests for it too.

Both urls are the same and lead to the same server/folder.
`git@github.com:facebook/docusaurus.git`
`ssh://git@github.com:22/facebook/docusaurus.git`

But I changed it to use the short url and keep it safe for everyone.